src/core/rattle.cpp

```diff
@@ -268,31 +272,33 @@ void apply_vel_corr() {
 }
 
 /**Put back the forces from r.p_old to f.f*/
-void revert_force() {
+void revert_force(const ParticleRange &particles,
+                  const ParticleRange &ghost_particles) {
   auto revert = [](Particle &p) {
     for (int j = 0; j < 3; j++)
       p.f.f[j] = p.r.p_old[j];
   };
 
-  for (auto &p : local_cells.particles())
+  for (auto &p : particles)
     revert(p);
 
-  for (auto &p : ghost_cells.particles())
+  for (auto &p : ghost_particles)
     revert(p);
 }
 
-void correct_vel_shake() {
+void correct_vel_shake(const ParticleRange &particles,
```

Comment:
Same comment as for `force_calc` applies, also the range should be reused withing thre function.